source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -616,10 +622,272 @@ namespace AdaptiveCards { namespace XamlCardRenderer
+        {
+            // Handle compact style
+            ComPtr<ISelector> selector;
+            THROW_IF_FAILED(inputItem->uiInputElement.As(&selector));
+
+            INT32 selectedIndex;
+            THROW_IF_FAILED(selector->get_SelectedIndex(&selectedIndex));
+
+            std::string choiceValue;
+            GetChoiceValue(choiceInput.Get(), selectedIndex, choiceValue);
+            jsonValue[idString] = choiceValue;
+        }
+        else
+        {
+            // For expanded style, get the panel children
+            ComPtr<IPanel> panel;
+            THROW_IF_FAILED(inputItem->uiInputElement.As(&panel));
+
+            ComPtr<IVector<UIElement*>> panelChildren;
+            THROW_IF_FAILED(panel->get_Children(panelChildren.ReleaseAndGetAddressOf()));
+
+            UINT size;
+            THROW_IF_FAILED(panelChildren->get_Size(&size));
+
+            if (isMultiSelect)
+            {
+                // For multiselect, gather all the inputs in a Json::arrayValue
+                Json::Value multiSelectValues(Json::arrayValue);
+                for (UINT i = 0; i < size; i++)
+                {
+                    ComPtr<IUIElement> currentElement;
+                    THROW_IF_FAILED(panelChildren->GetAt(i, &currentElement));
+
+                    boolean checkedValue = false;
+                    XamlHelpers::GetToggleValue(currentElement.Get(), &checkedValue);
+
+                    if (checkedValue)
+                    {
+                        std::string choiceValue;
+                        GetChoiceValue(choiceInput.Get(), i, choiceValue);
+                        multiSelectValues.append(Json::Value(choiceValue.c_str()));
+                    }
+                }
+                jsonValue[idString] = multiSelectValues;
+            }
+            else
+            {
+                // Look for the single selected choice
+                INT32 selectedIndex;
+                for (UINT i = 0; i < size; i++)
+                {
+                    ComPtr<IUIElement> currentElement;
+                    THROW_IF_FAILED(panelChildren->GetAt(i, &currentElement));
+
+                    boolean checkedValue = false;
+                    XamlHelpers::GetToggleValue(currentElement.Get(), &checkedValue);
+
+                    if (checkedValue)
+                    {
+                        selectedIndex = i;
+                        break;
+                    }
+                }
+                std::string choiceValue;
+                GetChoiceValue(choiceInput.Get(), selectedIndex, choiceValue);
+                jsonValue[idString] = choiceValue;
+            }
+        }
+    }
+
+    void SerializeInputs(
+        std::shared_ptr<std::vector<InputItem>> inputElements,
+        std::string* inputString)
+    {
+        Json::Value jsonValue;
+        for (std::vector<InputItem>::iterator inputItem = inputElements->begin(); inputItem != inputElements->end(); ++inputItem)
```

Comment:
>std::vector<InputItem>::iterator inputItem = inputElements->begin(); inputItem != inputElements->end(); ++inputItem [](start = 13, length = 115)

[nit] I usually prefer range-based for loops over iterators in cases where I go over an entire collection